pandas/core/window.py

```diff
@@ -0,0 +1,1339 @@
+
+            if values.ndim > 1:
+                result = np.apply_along_axis(calc, self.axis, values)
+            else:
+                result = calc(values)
+
+            if center:
+                result = self._center_window(result, window)
+
+            results.append(result)
+
+        return self._wrap_results(results, blocks, obj)
+
+class _Rolling_and_Expanding(_Rolling):
+
+    _shared_docs['count'] = """%(name)s count of number of non-NaN observations inside provided window."""
+    def count(self):
+        obj = self._convert_freq()
+        window = self._get_window()
+        window = min(window, len(obj)) if not self.center else window
+        try:
+            converted = np.isfinite(obj).astype(float)
+        except TypeError:
+            converted = np.isfinite(obj.astype(float)).astype(float)
+        result = self._constructor(converted,
+                                   window=window,
+                                   min_periods=0,
+                                   center=self.center).sum()
+
+        result[result.isnull()] = 0
+        return result
+
+    _shared_docs['apply'] = dedent("""
+    %(name)s function apply
+
+    Parameters
+    ----------
+    func : function
+    Must produce a single value from an ndarray input
+    *args and **kwargs are passed to the function""")
+
+    def apply(self, func, args=(), kwargs={}):
+        _level = kwargs.pop('_level',None)
+        window = self._get_window()
+        offset = _offset(window, self.center)
+        def f(arg, window, min_periods):
+            minp = _use_window(min_periods, window)
+            return algos.roll_generic(arg, window, minp, offset, func, args, kwargs)
+
+        return self._apply(f, center=False)
+
+    _shared_docs['sum'] = """%(name)s sum"""
+    def sum(self):
+        return self._apply('roll_sum')
+
+    _shared_docs['max'] = dedent("""
+    %(name)s maximum
+
+    Parameters
+    ----------
+    how : string, default max
+    Method for down- or re-sampling""")
+
+    def max(self, how='max'):
+        return self._apply('roll_max', how=how)
+
+    _shared_docs['min'] = dedent("""
+    %(name)s minimum
+
+    Parameters
+    ----------
+    how : string, default min
+    Method for down- or re-sampling""")
+
+    def min(self, how='min'):
+        return self._apply('roll_min', how=how)
+
+    _shared_docs['mean'] = """%(name)s mean"""
+    def mean(self):
+        return self._apply('roll_mean')
+
+    _shared_docs['median'] = dedent("""
+    %(name)s median
+
+    Parameters
+    ----------
+    how : string, default median
+    Method for down- or re-sampling""")
+
+    def median(self, how='median'):
+        return self._apply('roll_median_c', how=how)
+
+    _shared_docs['std'] = dedent("""
+    %(name)s standard deviation
+
+    Parameters
+    ----------
+    ddof : int, default 1
+    Delta Degrees of Freedom.  The divisor used in calculations
```

Comment:
the code above looks different than the current. maybe you have an older version?